# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gcc/gcc33/install_so.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./install-sh.orig	Sat Mar 23 08:04:55 2002
+++ ./install-sh	Sat Mar 23 08:10:11 2002
@@ -242,8 +242,12 @@
 
 # Now rename the file to the real destination.
 
-	$doit $rmcmd -f $dstdir/$dstfile &&
-	$doit $mvcmd $dsttmp $dstdir/$dstfile 
+	if echo $dstdir/$dstfile | grep -q 'lib/.*\.so' ; then
+		$doit $mvcmd $dsttmp $dstdir/$dstfile 
+	else
+		$doit $rmcmd -f $dstdir/$dstfile &&
+		$doit $mvcmd $dsttmp $dstdir/$dstfile 
+	fi
 
 fi &&
 
